packages/cdktf/lib/terraform-resource.ts

```diff
@@ -231,4 +252,64 @@ export class TerraformResource
       this.cdktfStack
     );
   }
+
+  private parentStackResourceTargets() {
+    return TerraformStack.of(this).moveTargets;
+  }
+
+  /**
+   * Moves this resource to the location denoted by moveTarget.
+   * @param moveTarget The previously set string moveTarget denoting the location of the move
+   * @param index Optional The index corresponding to the key the resource is to appear in the foreach of a resource to move to
+   */
+  public moveTo(moveTarget: string, index?: string | number) {
+    const stackMoveTargets = this.parentStackResourceTargets();
+    const resourceToMoveTo =
+      stackMoveTargets.getResourceAddressByTarget(moveTarget);
+    if (this.terraformResourceType !== resourceToMoveTo.terraformResourceType) {
+      throw new Error(
+        `The move target "${moveTarget}" corresponding to the resource of type ${resourceToMoveTo.terraformResourceType} to move to differs from the resource of type ${this.terraformResourceType} being moved from`
+      );
```

Comment:
I'd preface with a line like "You seem to move a resource to a different type" or something like that.

That said, I suspect there might be cases where this would actually work (i.e. if there are two resource types that target the same cloud vendor resource). However, I'd ignore that for now and call that out of scope.

Reply:
👍